### ISSUE TITLE
Incorrect suggestion (use map instead of zipWith)

### ISSUE THREAD

**Issue by zenhack** [ndmitchell/hlint#122]
When run on this file:

https://github.com/zenhack/phascal/blob/6084c676fcc3972d2020c26467af951ecb1b7add/src/Phascal/SymbolTable.hs

HLint makes the suggestion:

```haskell
src/Phascal/SymbolTable.hs:36:10: Error: Use map
Found:
  zipWith SymInfo [0 ..] (repeat ty)
Why not:
  map (\ x -> SymInfo [0 ..] ty) [0 ..]

1 suggestion
```

Which is incorrect -- and doesn't even type check.

**Comment by ndmitchell** [comment]
Thanks for the report. A bug in the `zipWith/repeat` rule, which I've now fixed. I've released 1.9.20 with the fix included.
